dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -454,37 +503,49 @@ public void validateVanityUrl(Contentlet contentlet) {
      * contents
      * @return A set of CachedVanityUrl, if null is because the cache needs to be initialized
      */
-    private Set<CachedVanityUrl> getVanityUrlBySiteAndLanguageFromCache(final String siteId,
+    private List<CachedVanityUrl> getVanityUrlBySiteAndLanguageFromCache(final String siteId,
             final long languageId,
             final Boolean includeSystemHost) {
 
         //First search in cache with the given site id
-        Set<CachedVanityUrl> foundVanities = CacheLocator.getVanityURLCache()
-                .getCachedVanityUrls(VanityUrlUtil.sanitizeSecondCacheKey(siteId, languageId));
+        List<CachedVanityUrl> foundVanities =
+                this.vanityUrlServices.getCachedVanityUrlList(siteId, languageId);
 
         //null means we need to initialize the cache for this site
-        if (null == foundVanities) {
-            return null;
-        }
-
-        if (includeSystemHost && !siteId.equals(Host.SYSTEM_HOST)) {
-            //Now search in cache with the SYSTEM_HOST
-            final Set<CachedVanityUrl> systemHostFoundVanities = CacheLocator.getVanityURLCache()
-                    .getCachedVanityUrls(
-                            VanityUrlUtil.sanitizeSecondCacheKey(Host.SYSTEM_HOST, languageId));
-
-            if (null != systemHostFoundVanities) {
-                foundVanities = ImmutableSet.<CachedVanityUrl>builder()
-                        .addAll(foundVanities)
-                        .addAll(systemHostFoundVanities)
-                        .build();
-            } else {
-                //This means we need to initialize the cache for the SYSTEM_HOST
-                foundVanities = null;
+        if (null != foundVanities) {
+
+            if (includeSystemHost && !siteId.equals(Host.SYSTEM_HOST)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)